compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -1598,9 +1599,18 @@ private <E extends BLangNode> E rewrite(E node, SymbolEnv env) {
     }
 
     private <E extends BLangNode> List<E> rewrite(List<E> nodeList, SymbolEnv env) {
-        for (int i = 0; i < nodeList.size(); i++) {
-            nodeList.set(i, rewrite(nodeList.get(i), env));
+        Queue<BLangSimpleVariableDef> previousQueue = this.annotationClosureReferences;
+        this.annotationClosureReferences = new LinkedList<>();
+        int size = nodeList.size();
+        for (int i = 0; i < size; i++) {
+            E node = rewrite(nodeList.remove(0), env);
+            Iterator<BLangSimpleVariableDef> iterator = annotationClosureReferences.iterator();
+            while (iterator.hasNext()) {
+                nodeList.add(rewrite((E) annotationClosureReferences.poll().var, env));
+            }
```

Comment:
This adds annotation closure calls as global variables, right? Ideally, we should be able to have them as local variables in the init method? Please create an issue.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/39575